.ci/Jenkinsfile

```diff
@@ -142,6 +148,40 @@ def cleanup(){
   unstash 'source'
 }
 
+def generateTestSignWithElasticStage() {
+  return {
+      INFRA_SIGNING_BUCKET_NAME = 'internal-ci-artifacts'
+      INFRA_SIGNING_BUCKET_CREDENTIALS = 'internal-ci-gcs-plugin'
+      INFRA_SIGNING_BUCKET_SUBFOLDER = "${env.REPO_NAME}/${env.BUILD_TAG}"
+      INFRA_SIGNING_BUCKET_PATH = "gs://${env.INFRA_SIGNING_BUCKET_NAME}/${env.INFRA_SIGNING_BUCKET_SUBFOLDER}"
+      INFRA_SIGNING_BUCKET_SUBFOLDER_SIGNED_ARTIFACTS = "${env.INFRA_SIGNING_BUCKET_SUBFOLDER}/signed-artifacts"
+      INFRA_SIGNING_BUCKET_SIGNED_ARTIFACTS_PATH = "gs://${env.INFRA_SIGNING_BUCKET_NAME}/${env.INFRA_SIGNING_BUCKET_SUBFOLDER_SIGNED_ARTIFACTS}"
+
+    withNode(labels: "ubuntu-20 && immutable", sleepMax: 20, forceWorkspace: true) {
+      cleanup()
+      dir("${BASE_DIR}"){
+        withMageEnv(){
+          sh(label: 'Build elastic-package',script: "make build")
+          sh(label: 'Prepare for tests (build zipped packages)',script: "make build-unsigned-zip-for-tests")
+          googleStorageUpload(bucket: env.INFRA_SIGNING_BUCKET_NAME,
+            credentialsId: env.INFRA_SIGNING_BUCKET_CREDENTIALS,
+            pathPrefix: 'build/integrations/',
+            pattern: 'build/integrations/*.zip',
+            sharedPublicly: false,
+            showInline: true)
+          build(wait: true, propagate: true, job: 'elastic+unified-release+master+sign-artifacts-with-gpg',
```

Comment:
Hey @v1v,

I think I have just understood the architecture used by the `apm-agent-php` to sign their artifacts. It's a bit different than I imagined :) Let me know if I'm correct - the signing procedure is executed in the internal-ci, but not only the signing step but the entire pipeline (see: [internal-ci](https://internal-ci.elastic.co/blue/organizations/jenkins/elastic%2Bapm-agent-php%2Brelease/detail/v1.3.1/1/pipeline/2530), [apm-ci](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-php%2Fapm-agent-php-mbp/detail/v0.3/1/pipeline/)).

Let's consider the elastic/integrations:
I was wondering if there is an option to support it without re-running the entire Integrations pipeline in the internal-ci, but just call the signing pipeline there?

Otherwise, I understand that we need a similar [JJB](https://github.com/elastic/infra/pull/23408/files) in the internal-ci for elastic/integrations, which will run just building (not testing), signing, and publishing to the Package Storage.

Reply:
Another thought:

is there an option to use the remote API? so that any Elastic pipeline (with auth) can trigger a build in the internal-ci instance?

Reply:
> is there an option to use the remote API? so that any Elastic pipeline (with auth) can trigger a build in the internal-ci instance?

IIRC, I think there is no such a integration, though let me dig a bit into what we discussed for the php agent in case those questions were also answered.

Reply:
Thanks! I'm afraid that it will force package owners to create artificial pipelines just to reach out to the internal-ci pipeline, which seems to be an unnecessary step if we have such an API.

Reply:
I'm afraid I could not find something that validates my hypothesis , so I just asked the `infra-automation` team, you are also poked in that particular conversation. To be clear, the conversation is outside of this thread, but I'll comment here the outcome